crates/bevy_ui/src/widget/text.rs

```diff
@@ -64,16 +64,20 @@ impl Measure for TextMeasure {
             AvailableSpace::MaxContent => self.info.max.x,
         });
 
-        height
-            .map_or_else(
-                || match available_width {
-                    AvailableSpace::Definite(_) => self.info.compute_size(Vec2::new(x, f32::MAX)),
-                    AvailableSpace::MinContent => Vec2::new(x, self.info.min.y),
-                    AvailableSpace::MaxContent => Vec2::new(x, self.info.max.y),
-                },
-                |y| Vec2::new(x, y),
-            )
-            .ceil()
+        let y = height.unwrap_or_else(|| {
+            if width.is_some() || matches!(available_width, AvailableSpace::Definite(_)) {
+                // given a definite width we have a definite height
+                self.info.compute_size(Vec2::new(x, f32::MAX)).y
+            } else {
+                match available_height {
+                    AvailableSpace::MinContent => self.info.min.y,
+                    AvailableSpace::MaxContent => self.info.max.y,
+                    AvailableSpace::Definite(_) => self.info.compute_size(Vec2::new(x, f32::MAX)).y,
+                }
```

Comment:
I think this should be changed back to `available_width`, which would allow the `available_width` case in the above PR to be removed.